File: LeetCode-Java/Search Insert Position.java
public class Solution {
    public int searchInsert(int[] A, int target) 
    {
        int start = 0, end = A.length - 1;
        int mid = -1;
        while(end >= start)
        {
            mid = (start + end) / 2;
            if(A[mid] == target) return mid;
            else if(A[mid] < target) start = mid + 1;
            else end = mid - 1;
        }
        return start;
       
    }
}

File: Leetcode-cpp/SearchInsertPosition.cpp
class Solution {
public:
    int searchInsert(int A[], int n, int target) {
       int head = 0,tail = n - 1;
       while(head < tail)
       {
           int mid = (head + tail) / 2;
           if(A[mid] == target) return mid;
           if(A[mid] < target)
           {
               head = mid + 1;
           }
           else
           {
               tail = mid - 1;
           }
       }
       if(A[head] < target) return head + 1;
       else return head;
    }
};
